nodejs/test/test-runner.ts

```diff
@@ -53,7 +53,7 @@ export function run(testDataFolder: string): void {
     // add cases
     describe(`${model}`, () => {
       let session: InferenceSession;
-      const skipModel = shouldSkipModel(model, ['cpu']);
+      const skipModel = shouldSkipModel(model, ['cpu'], modelPath);
       if (!skipModel) {
         before(async () => {
           session = await InferenceSession.create(modelPath);
```

Comment:
how about just put the try-catch check here? you can use a variable to indicate whether to run or to skip inside it()

putting check inside `shouldSkipModel()` may cause a longer test time as every model will be loaded twice, if `ALLOW_RELEASED_ONNX_OPSET_ONLY` is set

Reply:
Isn't it cleaner to have it in the utils so it's more easily re-used in other places and not just here? It's one aspect of checking whether the model should be skipped, so makes sense to me to be in shouldSkipModel.

---
In reply to: [485944450](https://github.com/microsoft/onnxruntime/pull/5096#discussion_r485944450) [](ancestors = 485944450)

Reply:
It is definitely cleaner to have it in the utils, it is best to have all the code to check whether the test should be skipped in 1 place. Moving it inside the async code and adding skip there has some issues mainly because mocha marks such tests as pending and not skipped.
 
Argument to remove loading model twice is compelling that is why I pushed a commit to give it a try... On my machine if I run all the tests then mocha hangs after all the tests are run however with a small number of tests it works just fine. Not sure if this is machine specific or because it thinks some tests are pending. I want to see what happens in the pipeline.

Reply:
CIs for nodejs pass. The issue I am seeing is specific to my machine because I see nodejs debugger hang too at odd places...
shouldSkipModel caters to filtering tests which are hardcoded to be skipped ... right now it is not trivial to move the check I am adding inside shouldSkipModel without avoiding a duplicate model load. So it is best to keep this separate for now.

Reply:
I am not sure how other tests deal with it (ALLOW_RELEASED_ONNX_OPSET_ONLY==1). If we need to figure out which tests should be skipped due to opset setting, in multiple places (nuget/python/c++), the best way is to use a script to generate a list so that we do this once and use it in all place.

Reply:
This is something we added recently after we decided to start picking up the commits from onnx master instead of release branch only (to increase development agility)
For other tests (onnx_tets_runner and onnx backend series) we have access to 1. domain version map from onnx and 2. the op schema which has op version  these 2 things make it easy to decide whether the test should be skipped without having to load the model twice. 
I don't agree with generating a list and referencing it in tests ... since we have all the info available to make a decision during runtime there is no need to keep this list around which will have to be regenerated everytime we move onnx commit etc.